### PR TITLE
Composer integration and bugfixes

### PR DESCRIPTION
**Pull request by noptic**
Hi your project is just what I was looking for, but I had to make some adjustments:
- composer now adds the commandline tool to vendor/bin
- commandline tool now includes vendor/autoload.php
- saveAs opens files in write mode.
- interfaces now support namespaces
- type hints work in namespaces
- interface metods are always public so I remoced the "public" modifier

### CODE REVIEW

File: bin/Distiller [hunk 1]
```diff
@@ -0,0 +1,13 @@
```
Comment (noptic) [inline bin/Distiller:0]: can be added to vendor/bin even if the php ini does not allow creating phar files

File: src/Distillate/Writer.php [hunk 2]
```diff
@@ -46,7 +46,12 @@ protected function writeString($string)
      */
     protected function writeInterfaceSignature($interfaceName, $extendingInterfaces = false)
     {
-        $this->writeString("interface $interfaceName");
```
Comment (noptic) [inline src/Distillate/Writer.php:49]: If the interface name contains a "\"  the interface is created in a namespace

File: src/Distillate/Writer.php [hunk 3]
```diff
@@ -72,7 +77,7 @@ protected function writeMethod(\ReflectionMethod $method)
     {
         $this->writeString(
             sprintf(
-            	'%s    public%sfunction %s(%s);',
+            	'%s    %sfunction %s(%s);',
```
Comment (noptic) [inline src/Distillate/Writer.php]: Interface methods are always public

File: src/Distillate/Writer.php [hunk 4]
```diff
@@ -118,7 +123,7 @@ protected function parameterToString(\ReflectionParameter $parameter)
         return trim(
             sprintf(
             	'%s%s %s$%s%s',
```
Comment (noptic) [inline src/Distillate/Writer.php:134]: If the interface is not in the global namespace the typehint must start with a "\"

File: src/Controller/CommandLine.php [hunk 5]
```diff
@@ -64,7 +64,7 @@ protected function applyOptions(array $cliArguments)
                 	break;
                 case 'saveAs':
                     $this->interfaceDistiller->$arg(
-                        new \SplFileObject(array_shift($cliArguments))
+                        new \SplFileObject(array_shift($cliArguments), 'w')
```
Comment (noptic) [inline src/Controller/CommandLine.php:67]: the default mode is 'r'.. omitting the parameter results in a exception

**Comment by gooh** [comment]
Hi. Thanks for all these improvements. Looks good. I am somewhat hesitant to merge the PR though because the merging will break the test suite.

File: src/Distillate/Writer.php (hunk 3 above)
Comment (gooh) [inline src/Distillate/Writer.php]: The Distiller does not follow a particular coding standard but PSR-2 suggests not to omit the visibility. For consistency, I'd prefer to keep the visibility.

**Comment by noptic** [comment]
I look into the test cases.
About the public modifier:
Sticking to PSR-2 is agood idead I´ll put it back in.

2013/3/13 gooh notifications@github.com

File: src/Distillate/Writer.php [hunk 6]
```diff
@@ -115,10 +128,11 @@ protected function methodParametersToString(\ReflectionMethod $method)
      */
     protected function parameterToString(\ReflectionParameter $parameter)
     {
+        $classPrefix = $this->inGlobalNamespace ? '': '\\';
```
Comment (noptic) [inline src/Distillate/Writer.php:131]: make sure the leading \ is not added in global namespace

**Comment by noptic** [comment]
All methods are declared public again.
Unit tests run under PHP 5.3.3.
My IDE is cloud9 and as far as I know there is no way to upgrade it to PHP 5.4.

**Comment by gooh** [comment]
Thanks! Much appreciated.

**Pull request merged**